src/containerapp/azext_containerapp/_ssh_utils.py

```diff
@@ -0,0 +1,57 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=logging-fstring-interpolation
+# pylint: disable=possibly-used-before-assignment
+
+from azure.cli.command_modules.containerapp._ssh_utils import WebSocketConnection, SSH_TERM_RESIZE_PREFIX, \
+    SSH_DEFAULT_ENCODING, read_ssh
+from azure.cli.core.azclierror import ValidationError, CLIInternalError
+
+from knack.log import get_logger
+
+from azext_containerapp._clients import ContainerAppPreviewClient
+
+logger = get_logger(__name__)
+
+
+class DebugWebSocketConnection(WebSocketConnection):
+    def __init__(self, cmd, resource_group_name, name, revision, replica, container):
+        try:
+            super(DebugWebSocketConnection, self).__init__(cmd, resource_group_name,
+                                                           name, revision, replica, container, "")
+        except Exception as e:
+            stringErr = str(e)
+            if "Handshake status 400 Bad Request" in stringErr:
+                raise CLIInternalError("Debug Console feature is not allowed for the subscription.")
+            raise e
+
+    @classmethod
+    def _get_logstream_endpoint(cls, cmd, resource_group_name, name, revision, replica, container):
```

Comment:
why are we overriding this class method with no implementation here..?

Reply:
This method from super class `WebSocketConnection` will request to get replica. override it to reduce a request.